pkg/bootstrap/init.go

```diff
@@ -251,51 +200,33 @@ func yamlToObject(yamlContent []byte) (*unstructured.Unstructured, error) {
 	return obj, nil
 }
 
-func createOrUpdateResourcesFromDir(dir string, client client.Client, reader client.Reader) error {
-	files, err := ioutil.ReadDir(dir)
+func createOrUpdateResourcesFromAnnotation(client client.Client, reader client.Reader) error {
+	deploy, err := getDeployment(reader)
 	if err != nil {
 		return err
 	}
-
-	var yamlFiles []string
-
-	for _, file := range files {
-		if filepath.Ext(file.Name()) == ".yaml" {
-			yamlFiles = append(yamlFiles, file.Name())
-		}
-	}
-
-	for _, file := range yamlFiles {
-		yamlContent, err := ioutil.ReadFile(filepath.Join(dir, file))
-		if err != nil {
-			return err
-		}
-		if err := createOrUpdateFromYaml(yamlContent, client, reader); err != nil {
-			return err
+	annotations := deploy.Spec.Template.GetAnnotations()
+	for _, res := range CsResources {
+		if res == "extraResources" {
```

Comment:
You can separate this if into a function to reduce the nested levels and make the code more readable.

Reply:
Splited